### ISSUE TITLE
Project, Workspaces, & Session Management

### ISSUE THREAD

**Issue by UsuallySleebee** [NiamhFerns/Nekovim#10]
Config should allow for creation, management, and loading of workspaces either per project or workspaces unrelated to a project. 

Should also track session in those workspaces so that reloading the workspace will reload your previous session as it would with something like a Jetbrains IDE. Sessions should automatically reload when reopening a project.

**Features:**
- [x] Open a project through project.nvim.
- [ ] On opening a project show...
    - *No Saved Sessions*: the default project.nvim telescope fuzzy file picker.
    - *Auto-Saved Session*: automatically open the last auto-save.
    - *Saved Session w. No Auto-Save*: open session picker.
- [ ] Should save a project session automatically on closing neovim (auto-save).
- [ ] Should save a project session automatically on switching projects (auto-save).
- [ ] Should be able to manually save a project session with \<leader\>pS
- [ ] Should be able to manually save a general session with \<leader\>pG
- [ ] Should be able to browse only sessions related to a project with \<leader\>ps
- [ ] Should be able so browse all sessions with telescope with \<leader\>pg.
- [ ] Only one auto-save should exist per project.
- [ ] Sessions should not save certain buffers (Neotree, Nvim-tree, Toggleterm, etc.)

**UPDATE:** Still undecided and need to look at how creating plugins in Neovim even works but this feature pretty much requires me to create a plugin to do what I want.

**Comment by UsuallySleebee** [comment]
Best choices most likely 
- [Auto Session](https://github.com/rmagatti/auto-session): Supports hooks for loading sessions as well as saving/loading them manually. Sessions saved to configurable directory.
- [Session Lens](https://github.com/rmagatti/session-lens): Telescope integration for Auto Session.

**Comment by UsuallySleebee** [comment]
Currently waiting on https://github.com/rmagatti/auto-session/pull/161 to merge.

**Comment by UsuallySleebee** [comment]
After some further research... getting this exact functionality is going to be a real pain and will probably have to be implemented as an add-on to another plugin.

I'm thinking of just implementing this as an add-on for Project.nvim with Project.nvim opening the fuzzy file finder if there are no sessions and opening a session lens type telescope finder that will show only local sessions. 

Should give the option to add *global* sessions as well where the session and session name are stored in a place of the user's choice (`$HOME/.local/share/nvim/sessions` by default.

**Comment by UsuallySleebee** [comment]
New plan: just use mini.sessions and find a way to integrate it with project.nvim. You can make an addition to the telescope plugin potentially.

**Comment by UsuallySleebee** [comment]
New new plan, mini.sessions straight up didn't work. Using folke/persistence.nvim instead and now project saving and loading works on exiting nvim and on opening a project. Works on switching projects but doesn't save the previous session before switch.
